### PR TITLE
Implemented if `indices_sorted` is True

### PR DESCRIPTION
**Pull request by Aniketsy**
Includes a function to determine if a 1D or 2D array of indices is sorted along axis =0,
which is important for certain sparse matrix operations and optimizations.

If this approach is not correct or can be improved, Please let me know.

### CODE REVIEW

**Comment by jakevdp** [comment]
Thanks for looking at this – I notice the function you added is not used anywhere. Is there additional code that you forgot to commit?

**Comment by Aniketsy** [comment]
https://github.com/jax-ml/jax/blob/main/jax/experimental/sparse/bcoo.py

`def` _unique_indices_unbatched(indices, *, shape, return_inverse=False,
                              return_index=False, return_true_size=False):
  props = _validate_bcoo_indices(indices, shape)
  if props.n_sparse == 0:
    nse = 1
    indices_out = jnp.zeros_like(indices, shape=(nse, 0))
    out = (indices_out,)
    if return_index:
      out = (*out, jnp.zeros(nse, dtype='int32'))
    if return_inverse:
      out = (*out, jnp.zeros(nse, dtype='int32'))
    if return_true_size:
      out = (*out, nse)
    return out[0] if len(out) == 1 else out
  fill_value = jnp.expand_dims(jnp.array(shape[:props.n_sparse], dtype=indices.dtype), (0,))
  out_of_bounds = (indices >= fill_value).any(-1, keepdims=True)
  indices = jnp.where(out_of_bounds, fill_value, indices)
  # TODO: check if `indices_sorted` is True.
  out = _unique(indices, axis=0, return_inverse=return_inverse, return_index=return_index,
                return_true_size=return_true_size, size=props.nse, fill_value=fill_value)
  if return_inverse:
    idx = 2 if return_index else 1
    out = (*out[:idx], out[idx].ravel(), *out[idx + 1:])
  if return_true_size:
    nse = out[-1]
    nse = nse - (indices == fill_value).any().astype(nse.dtype)
    out = (*out[:-1], nse)
  return out

In this implementation, I attempted to complete the TODO section. Alternatively, instead of defining a separate function, we could directly use a conditional statement to achieve the same functionality.

**Comment by jakevdp** [comment]
OK - so for more context on that TODO, the BCOO object has a flag that tracks whether indices are sorted. `_unique` can be computed more efficiently if it is known that the input is sorted, and so we could make some marginal efficiency improvements by piping that flag through and adding an option to the internal `_unique` function.

That said, we're not investing much in the BCOO code anymore, so I wouldn't suggest taking the time to do all that. I'm going to close this PR since it's probably not a change we'd want to merge.

**Comment by Aniketsy** [comment]
Thanks for the feedback and for explaining the context around the sorted flag and _unique—that really helped me understand the reasoning better. I totally understand the decision to not move forward with changes to BCOO.

I'm currently exploring ways to contribute more meaningfully to the project, especially with the hope of participating in Google Summer of Code (GSoC). If there are any areas, open issues, or components where beginner-friendly contributions would be helpful, I’d really appreciate your suggestions on where I could focus my efforts.

Thanks again for your time and guidance!
